### PR TITLE
Add async gpu wait interface

### PR DESCRIPTION
**Pull request by rabbitsaviola**

### CODE REVIEW

File: src/lv_core/lv_refr.c [hunk 1]
```diff
@@ -741,6 +741,13 @@ static void lv_refr_vdb_flush(void)
 
     /*Flush the rendered content to the display*/
     lv_disp_t * disp = _lv_refr_get_disp_refreshing();
+#if LV_USE_GPU
```
Comment (kisvegabor) [inline src/lv_core/lv_refr.c]: It's a little bit misleading (will be updated) but `LV_USE_GPU` enables only the 2 GPU callbacks in the display driver. Independently DMA2D (ST) os PXP (NXP) GPU can be enabled.

So these `#if LV_USE_GPU` statements are not required.

File: src/lv_core/lv_refr.c [hunk 2]
```diff
@@ -741,6 +741,13 @@ static void lv_refr_vdb_flush(void)
...
+    if (disp->driver.gpu_wait_cb)
```
Comment (kisvegabor) [inline src/lv_core/lv_refr.c]: Just to follow the coding style please write it like this
```c
if (disp->driver.gpu_wait_cb) disp->driver.gpu_wait_cb(&disp->driver);
```

**Review by kisvegabor — COMMENTED** [review]
Thank you very much. It looks good, only two minor things.

File: src/lv_core/lv_refr.c (hunk 1 above)
Comment (rabbitsaviola) [reply src/lv_core/lv_refr.c]: Do you mean DMA2D might be used even though `LV_USE_GPU` is disabled? Our solution implements two gpu callbacks to accelerate rendering using the GPU just like DMA2D.

File: src/lv_core/lv_refr.c (hunk 2 above)
Comment (rabbitsaviola) [reply src/lv_core/lv_refr.c]: ok, I'll change it.

File: src/lv_core/lv_refr.c (hunk 1 above)
Comment (kisvegabor) [reply src/lv_core/lv_refr.c]: > Do you mean DMA2D might be used even though LV_USE_GPU is disabled?

Yes, the GPU callbacks in the display driver have priority. If they are set LVGL uses them. If not set LVGL tries DMA2D, if it's not present or can't be used in the a given case, software rendering is applied. 


> Our solution implements two gpu callbacks to accelerate rendering using the GPU just like DMA2D.

You can simply remove the `#if LV_USE_GPU` guards as the `gpu_wait_cb` is useful for DMA2D too.

---

Probably `LV_USE_GPU` should be removed from LVGL as it causes confusion and saves only a very little RAM.

File: src/lv_hal/lv_hal_disp.h [hunk 3]
```diff
@@ -111,7 +111,11 @@ typedef struct _disp_drv_t {
      * User can execute very simple tasks here or yield the task */
     void (*wait_cb)(struct _disp_drv_t * disp_drv);
 
+    /** OPTIONAL: called to check whether gpu is idle */
```
Comment (kisvegabor) [inline src/lv_hal/lv_hal_disp.h]:
```suggestion
    /** OPTIONAL: called to wait while the gpu is working */
```

**Comment by kisvegabor** [comment]
Thanks for the update. I've added a minor suggestion about a comment and update the CHANGELOG.

**Comment by kisvegabor** [comment]
Great, thank you very much!

Merged!

**Pull request merged**